client/rpc/src/system/mod.rs

```diff
@@ -91,21 +94,41 @@ impl<B: traits::Block> SystemApi<B::Hash, <B::Header as HeaderT>::Number> for Sy
 		Receiver(Compat::new(rx))
 	}
 
-	fn system_peers(&self) -> Receiver<Vec<PeerInfo<B::Hash, <B::Header as HeaderT>::Number>>> {
+	fn system_peers(&self)
+		-> Compat<BoxFuture<'static, rpc::Result<Vec<PeerInfo<B::Hash, <B::Header as HeaderT>::Number>>>>>
+	{
+		if let Err(err) = self.deny_unsafe.check_if_safe() {
```

Comment:
I would suggest adding helper method that will contain this check and takes closure executed only if check passes

The check will be in one place then

Reply:
The error type is different for e.g. System and State API - should I add a separate helper method for each API?

The check is used to short-circuit, so it'd have to be a macro; I can't use `?` with a plain helper function since I'd have to materialize an `Ok` value if this succeeds. Is that still okay?